code/modules/modular_computers/file_system/programs/radar.dm

```diff
@@ -65,7 +82,37 @@
   *
 */
 /datum/computer_file/program/radar/proc/track()
-	return
+	var/atom/movable/signal = find_atom()
+	if(!trackable(signal))
+		return
+
+	var/turf/here_turf = (get_turf(computer))
+	var/turf/target_turf = (get_turf(signal))
+	var/userot = FALSE
+	var/rot = 0
+	var/pointer="crosshairs"
+	var/locx = (target_turf.x - here_turf.x) + 24
+	var/locy = (here_turf.y - target_turf.y) + 24
+
+	if(get_dist_euclidian(here_turf, target_turf) > 24)
+		userot = TRUE
+		rot = round(Get_Angle(here_turf, target_turf))
+	else
+		if(target_turf.z > here_turf.z)
+			pointer="caret-up"
+		else if(target_turf.z < here_turf.z)
+			pointer="caret-down"
+
+	var/list/trackinfo = list(
+		locx = locx,
+		locy = locy,
+		userot = userot,
+		rot = rot,
+		arrowstyle = arrowstyle,
+		color = pointercolor,
+		pointer = pointer,
+		)
```

Comment:
You might want to use string indexes, such as `"rot" = rot` and `"userot" = userot` else you'll have odd elements like `0 = 0` and whatnot, making this into an associative list which you can only access directly.

Reply:
Done.